### ISSUE TITLE
Consider tricks to shrink the versions table

### ISSUE THREAD

**Issue by simonw** [simonw/git-history#12]
With many copies of the row data and foreign keys that are string hashes it gets pretty big!

**Comment by simonw** [comment]
For the commits table I feel like I want a version of `table.lookup()` that can be passed additional columns to populate only if the record does not exist yet.

**Comment by simonw** [comment]
Using the Act Blue example from #17 I'm going to see how much of a size difference it makes to store integer IDs as the foreign keys in `item_versions` to commits and items as opposed to storing hashes.

**Comment by simonw** [comment]
Running current main like this:

    git-history file /tmp/actblue.db /tmp/actblue-ticker-tracker/sitewide_raised_amount.txt \
      --repo /tmp/actblue-ticker-tracker \
      --convert '[{"id": 1, "value": int(content)}]' --id id

And then running `sqlite-utils vacuum /tmp/actblue.db` gives this:
```
% ls -lah /tmp/actblue.db         
-rw-r--r--@ 1 simon  wheel   8.1M Nov 18 21:57 /tmp/actblue.db
```

**Comment by rdmurphy** [comment]
> With many copies of the row data and foreign keys that are string hashes it gets pretty big!

Yeah I was gonna submit a separate issue for this. 😅 I pointed `git-history` at the CSV in the ActBlue repo and ended up canceling it once it reached 4 GB. Might be an extreme case, though!

**Comment by simonw** [comment]
After much fiddling around I produced a version of the database using integer primary keys / foreign keys instead:

```
% ls -lah /tmp/actblue2.db            
-rw-r--r--  1 simon  wheel   4.6M Nov 18 22:51 /tmp/actblue2.db
```
Almost half the size. This optimization looks worthwhile!

**Comment by simonw** [comment]
Implemented this in #20.
